src/mapreduce.jl

```diff
@@ -302,8 +302,8 @@ end
 
     return quote
         @_inline_meta
-        T = typeof(one(eltype(a)) - one(eltype(a)))
-        @inbounds return similar_type(a, T, Size($Snew))(tuple($(exprs...)))
+        elements = tuple($(exprs...))
+        @inbounds return similar_type(a, eltype(elements), Size($Snew))(elements)
```

Comment:
`eltype(::Tuple)` uses a recursive algorithm with `promote_typejoin`. Not sure how it behaves with long tuples? (Or mixed types?)

Way way way way way way way back we had to use our own `promote_tuple_eltype` implementation to support long tuples gracefully.

Reply:
Well today in 1.4.0 we have

```
@inferred(diff(rand(SVector{100,SVector{2,Int}})))
```

being ok so perhaps this is fine?

Would it help if I added that to the tests?

Reply:
I guess? It’s hard to know what people expect us to do with their length-100 static arrays...

Reply:
Torture the compiler if it ends up in a fast code and continue doing so until it brings everything down? :trollface:  (That's just my impression. Honestly, I don't know.)

Reply:
Yeah, I think that's accurate :-D